sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -340,3 +340,28 @@ case class BroadcastPartitioning(mode: BroadcastMode) extends Partitioning {
     case _ => false
   }
 }
+
+/**
```

Comment:
Why we need to add a new `Partitioning ` ?

Reply:
`CoalescedHashPartitioning` can satisfy the `ClusteredDistribution` because a skew join may match the case which contains Aggregation (non-skew side). So `UnknownPartitioning` cannot satisfy `ClusteredDistribution` and add an additional shuffle.

Reply:
Hi @JkSelf I will provide another approach that removes this `CoalescedHashPartitioning` and simplify the code. But current implementation with `CoalescedHashPartitioning` might be more general for more cases.